app/src/main/java/edu/wpi/first/shuffleboard/app/plugin/PluginLoader.java

```diff
@@ -196,11 +404,22 @@ private void tryRestoreSource(Widget widget, DestroyedSource destroyedSource) {
    * Unloads a plugin.
    *
    * @param plugin the plugin to unload
+   *
+   * @return true if the plugin was unloaded, false if not. This can occur if the plugin was not loaded to begin with.
    */
   @SuppressWarnings("unchecked")
-  public void unload(Plugin plugin) {
+  public boolean unload(Plugin plugin) {
```

Comment:
What if a user tries to unload a plugin that another plugin has a dependency on?

Reply:
All dependent plugins are unloaded first.